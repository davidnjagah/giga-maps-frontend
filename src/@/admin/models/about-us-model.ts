import { combine, createEvent, createStore, sample } from "effector";

import { $notification } from "~/@/common/Toast/toast.model";
import { APIListType } from "~/api/types";
import { setPayload } from "~/lib/effector-kit";

import { aboutUsFormList, defaultAboutUsList } from "../constants/aboutus.constant";
import { getAboutusContentFx, getImagesListFx, updateAboutusContentFx } from "../effects/about-us-fx";
import { AboutusFormType, AboutUsImage } from "../types/about-us.type";
import { jsonToMapping, updateContentItemPaths, updateItemContentAndPath } from '../utils/aboutus.util';



export const onChangeAdminAboutUsTab = createEvent<number>();
export const $adminAboutUsTab = createStore(0)
$adminAboutUsTab.on(onChangeAdminAboutUsTab, setPayload);


export const $imageList = createStore<APIListType<AboutUsImage> | null>(null);
$imageList.on(getImagesListFx.doneData, setPayload);

export const onRemoveContentItem = createEvent<{ type: string, path: string; index: number; }>();
export const $formItemList = createStore<AboutusFormType[]>(aboutUsFormList);
export const onAddNewContent = createEvent<{ type: string; path: string }>();
$formItemList.on(onAddNewContent, (state, payload) => updateItemContentAndPath(state, payload, 1));


export const onSubmitUpdateFormList = createEvent<SaveFormListType>();
export const $saveFormList = createStore(defaultAboutUsList);
export const $aboutUsContent = createStore<SaveFormListType>([])
export const $aboutusDataAvailable = $saveFormList.map(items => items.some((item) => item.id))
export type SaveFormListType = ReturnType<typeof $saveFormList.getState>
const updateSaveFormList = (state: SaveFormListType, payload: SaveFormListType) => {
  if (payload?.length) {
    const mappedJson = jsonToMapping(payload, JSON.parse(JSON.stringify(state)));
    return mappedJson;
  }
  return state;
}
$aboutUsContent.on(getAboutusContentFx.doneData, setPayload)
$saveFormList.on(getAboutusContentFx.doneData, updateSaveFormList);
$saveFormList.on(onSubmitUpdateFormList, updateSaveFormList);

export const onChangeFormField = createEvent<{ path: string; value: string; type: string }>();

$saveFormList.on(onChangeFormField, (state, payload) => {
  const { path, value, type } = payload;
  const newState = state.map((item) => {
    if (item.type === type) {
      item[path] = value;
    }
    return item;
  });
  return newState;
})

// on remove content item
sample({
  clock: onRemoveContentItem,
  source: combine({ saveFormList: $saveFormList, formItemList: $formItemList }),
  fn: ({ saveFormList, formItemList }, { type, index, path }) => {
    const formItem = formItemList.find((item) => item.type === type);
    if (!formItem) return saveFormList;
    return saveFormList.map((item) => {
      if (item.type !== type) {
        return item;
      }
      Object.keys(item).forEach((key) => {
        const currentItem = `${path}.${index}`;
        if (key.startsWith(currentItem)) {
          delete item[key];
        }
      });
      return !!paths ? {
        ...item,
      } : item;
    });
  },
  target: $saveFormList
})

const getContentByPath = (item: Record<string, any>, path: string) => {
  const keys = path.split(".");
  return keys.reduce((obj, key) => obj?.[key], item);
}
// add new item on new data;
sample({
  clock: getAboutusContentFx.doneData,
  source: combine({ formItemList: $formItemList }),
  fn: ({ formItemList }, aboutusContentResponse) => {
    if (aboutusContentResponse.length > 0) {
      const newState = formItemList.map((matchedItem) => {
        const currentItem = aboutusContentResponse.find(item => item.type === matchedItem.type);
        const currentContentLength = currentItem?.content?.length || 0;
        const matchContentLength = matchedItem?.content?.items.length ?? 0;
        let newItemContent = matchedItem.content;
        if (currentItem && matchedItem.content && currentContentLength > matchContentLength) {
          const newCount = currentContentLength - matchContentLength;
          newItemContent = {
            ...matchedItem.content,
            items: updateContentItemPaths(matchedItem.content.items, matchedItem.content.path, newCount),
          };
        }
        return {
          ...matchedItem,
          fields: matchedItem.fields.map((field) => {
            const { path } = field;
            if (currentItem && path.startsWith('content') && field?.items?.length) {
              const getCurrentFieldContent = getContentByPath(currentItem, path)
              const count = getCurrentFieldContent?.length - field?.items?.length || 0;
              return {
                ...field,
                items: updateContentItemPaths(field.items, path, count),
              }
            }
            return field;
          }),
          content: newItemContent
        };
      });
      return newState;
    }
    return formItemList;
  },
  target: $formItemList
})

// sync content length
sample({
  clock: onRemoveContentItem,
  source: combine({ formItemList: $formItemList }),
  fn: ({ formItemList }, { type, index, path }) => {
    const isContentType = path === 'content';
    const newState = formItemList.map((section) => {
      if (section.type === type) {
        section = {
          ...section,
        }
        if (isContentType) {
          section.content = section.content ? {
            ...section.content,
            items: section?.content?.items.filter((_, i) => i !== index)
          } : undefined
        } else if (path.startsWith('content')) {
          section.fields = section.fields.map((field) => {
            if (field.path === path) {
              return {
                ...field,
                items: field?.items?.filter((_, i: number) => i !== index)
              }
            }
            return field;
          })
        }
      }
      return section;
    });
    return newState;
  },
  target: $formItemList
})

sample({
  clock: updateAboutusContentFx.done,
  fn: () => ({
    title: `Aboutus content saved.`,
    kind: 'success',
    subtitle: ''
  }),
  target: $notification
})