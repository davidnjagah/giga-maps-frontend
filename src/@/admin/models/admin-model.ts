import { createEvent, createStore, sample } from "effector";

import { setPayload } from "~/lib/effector-kit";

import { getAppConfigValuesFx, getInvalidateCacheFx } from "../effects/admin-main-fx";
import { ApiConfig, InvalidateCache } from "../types/giga-layer.type";

export const getAppConfigValues = createEvent<void>()
export const $appConfigValues = createStore<ApiConfig | null>(null)
$appConfigValues.on(getAppConfigValuesFx.doneData, setPayload);


sample({
  clock: getAppConfigValues,
  source: $appConfigValues,
  filter: (values: ApiConfig | null) => {
    return !values;
  },
  target: getAppConfigValuesFx,
})

export const $inValidateCacheResponse = createStore<InvalidateCache | null>(null)
$inValidateCacheResponse.on(getInvalidateCacheFx.doneData, setPayload);