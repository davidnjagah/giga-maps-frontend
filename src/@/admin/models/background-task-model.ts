import { createStore, sample } from "effector";
import { createGate } from "effector-react";

import { $notification } from "~/@/common/Toast/toast.model";
import { APIListType } from "~/api/types";
import { setPayload } from "~/lib/effector-kit";

import { defaultBackgroundTask } from "../constants/background-task.constant";
import { deleteBackgroundTaskFx, getBackgroundTaskIdFx, getBackgroundTaskListFx } from "../effects/background-task-fx";
import { BackgroundTask } from "../types/background-task.type";



export const $backgroundTaskList = createStore<APIListType<BackgroundTask> | null>(null);
$backgroundTaskList.on(getBackgroundTaskListFx.doneData, setPayload);

export const BackgroundTaskGate = createGate<{ page: number; }>();
export const $backgroundTaskPageNo = BackgroundTaskGate.state.map((state) => state.page);

export const $formBackgroundTask = createStore(defaultBackgroundTask);
$formBackgroundTask.on(getBackgroundTaskIdFx.doneData, setPayload)



sample({
  clock: deleteBackgroundTaskFx.done,
  fn: ({ params }) => ({
    title: `Record Deleted.`,
    kind: 'success',
    subtitle: ''
  }),
  target: $notification
})