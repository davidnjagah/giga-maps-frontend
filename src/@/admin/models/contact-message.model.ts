import { createStore, sample } from "effector";

import { $notification } from "~/@/common/Toast/toast.model";
import { APIListType } from "~/api/types";
import { setPayload } from "~/lib/effector-kit";

import { defaultContactMesaage } from "../constants/background-task.constant";
import { deleteContactMessageFx, getContactMessageIdFx, getContactMessageListFx } from "../effects/contact-message-fx";
import { ContactMessage } from "../types/contact-message.type";



export const $constactMessageList = createStore<APIListType<ContactMessage> | null>(null);
$constactMessageList.on(getContactMessageListFx.doneData, setPayload);

export const $formContactMessage = createStore(defaultContactMesaage);
$formContactMessage.on(getContactMessageIdFx.doneData, setPayload)


sample({
  clock: deleteContactMessageFx.done,
  fn: ({ params }) => ({
    title: `Record Deleted.`,
    kind: 'success',
    subtitle: ''
  }),
  target: $notification
})