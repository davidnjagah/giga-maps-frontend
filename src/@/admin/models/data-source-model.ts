import { $notification, onCreateNotification } from '~/@/common/Toast/toast.model';
import { combine, createEvent, createStore, merge, sample } from "effector";

import { setPayload, setPayloadResults } from "~/lib/effector-kit";

import { deleteSchoolMasterFx, getCountryDataSourceFx, getSchoolMasterListFx, publishAllSchoolMasterFx, publishSchoolMasterFx } from "../effects/data-source.fx";
import { previewDataSourceFx } from "../effects/preview-data-source-fx";
import { DataSourceType } from "../types/data-source.type";
import { $adminMap } from "~/@/common/admin-map-preview/admin-map.model";
import { deleteSourceAndLayers } from "~/@/map/utils";
import { CountryListType } from '~/@/api-docs/types/country-list.type';

export const $schoolMasterCount = createStore(0)
$schoolMasterCount.on(getSchoolMasterListFx.doneData, (_, payload) => payload?.count);

export const $schoolMasterData = createStore<DataSourceType[]>([])
$schoolMasterData.on(getSchoolMasterListFx.doneData, setPayloadResults);

export const clearDataSourcePreview = createEvent();

export const onPreviewDataSource = createEvent<DataSourceType[]>()

export const $countryDataSourceList = createStore<CountryListType[]>([]);
$countryDataSourceList.on(getCountryDataSourceFx.doneData, setPayload);
sample({
  clock: onPreviewDataSource,
  source: combine({ map: $adminMap }),
  filter: ({ map }) => {
    return !!map;
  },
  fn: ({ map }, mapData) => {
    return { map, mapData }
  },
  target: previewDataSourceFx
})

sample({
  clock: clearDataSourcePreview,
  source: $adminMap,
  fn: (map) => {
    if (map) {
      deleteSourceAndLayers({ map });
    }
    return null;
  }
})

sample({
  clock: merge([publishAllSchoolMasterFx.doneData, publishSchoolMasterFx.doneData]),
  fn: () => {
    return onCreateNotification({
      title: 'School master data',
      subtitle: 'Published selected records.',
      kind: 'info',
    })
  },
  target: onCreateNotification
})
sample({
  clock: deleteSchoolMasterFx.doneData,
  fn: () => {
    return onCreateNotification({
      title: 'School master data',
      subtitle: 'Discarded selected records.',
      kind: 'info',
    })
  },
  target: onCreateNotification
})