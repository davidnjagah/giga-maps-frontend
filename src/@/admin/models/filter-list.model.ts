import { createEvent, createStore, sample } from "effector";
import { filterColumnListFx, getFilterListFx } from "../effects/filter-fx";
import { setPayloadResults } from "~/lib/effector-kit";
import { FilterConfiguration, FilterListType } from "../types/filter-list.type";

export const $filterListResponse = createStore<FilterListType[]>([]);
export const $filterListCount = createStore(0);
$filterListResponse.on(getFilterListFx.doneData, setPayloadResults);
$filterListCount.on(getFilterListFx.doneData, (_, response) => response?.count || 0);

export const $filterColumnList = createStore<FilterConfiguration[]>([]);
$filterColumnList.on(filterColumnListFx.doneData, setPayloadResults);

export const onGetFilterList = createEvent<{ page: number; pageSize: number; }>();

sample({
  clock: onGetFilterList,
  target: getFilterListFx,
})