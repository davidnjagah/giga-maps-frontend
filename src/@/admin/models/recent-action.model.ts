import { createStore } from "effector";

import { setPayload, setPayloadResults } from "~/lib/effector-kit";

import { getRecentActionListFx } from "../effects/recent-action-fx";
import { RecentAction } from "../types/recent-action.type";
import { APIListType } from "~/api/types";



export const $recentActionList = createStore<APIListType<RecentAction> | null>(null);
$recentActionList.on(getRecentActionListFx.doneData, setPayload);

