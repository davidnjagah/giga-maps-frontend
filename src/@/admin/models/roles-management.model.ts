import { createEvent, restore } from "effector";

export const setShowduplicateRecordWarning = createEvent<boolean>()
export const $duplicateRecordWarning = restore(setShowduplicateRecordWarning, false);