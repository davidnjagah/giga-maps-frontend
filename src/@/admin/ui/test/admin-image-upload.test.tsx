import { fireEvent, render, screen } from "@testing-library/react"
import { createEvent } from "effector"

import { aboutUsImageList } from "~/tests/data/admin-about-us-image-list";
import { testWrapper } from "~/tests/jest-wrapper"

import { $imageList } from "../../models/about-us-model";
import ListImage from "../about-us/about-us-image-upload/list-image"
import ModalUploadImageContent from "../about-us/about-us-image-upload/modal-upload-image-content";

const setImageList = createEvent();
$imageList.on(setImageList, (_, payload) => payload)

describe("About Us image upload", () => {

  test("Render MainAboutUsView", () => {
    setImageList(aboutUsImageList)
    const { asFragment } = render(testWrapper(<ListImage />))
    expect(asFragment()).toMatchSnapshot();
  })

  test("Click on delete icon", () => {
    const handleClick = jest.fn();
    const { getByTestId } = render(testWrapper(<ListImage onClick={handleClick} />));
    const button = getByTestId(`delete-about-us-image${aboutUsImageList?.results[0].id}`);
    fireEvent.click(button);
    expect(screen.getByText('Delete role -')).toBeInTheDocument();
    expect(screen.getByText('Are you sure?')).toBeInTheDocument();
    const parentElement = getByTestId(`admin-about-us-delete-image`);
  })

  test("Click on yes to delete image", () => {
    const handleClick = jest.fn();
    const { getByTestId } = render(testWrapper(<ListImage onClick={handleClick} />));
    const button = getByTestId(`delete-about-us-image${aboutUsImageList?.results[0].id}`);
    fireEvent.click(button);
    render(<ListImage />)
    const parentElement = getByTestId(`admin-about-us-delete-image`);
    const childElement = parentElement.getElementsByClassName('cds--actionable-notification__action-button')
    const buttonArray = Array.from(childElement);
    expect(buttonArray.length).toBeGreaterThan(0);
    buttonArray.forEach(button => {
      fireEvent.click(button);
    });
  })

  test("Render ModalUploadImageContent", () => {
    render(<ModalUploadImageContent setOpen={true} />)
  })

  test("Click submit", () => {
    const handleClick = jest.fn();
    const { getByTestId } = render(<ModalUploadImageContent setOpen={handleClick} />)
    // const nameInput = getByTestId('image-upload-name');
    // fireEvent.change(nameInput, { target: { value: 'Test Name' } });
    const file = new File(['test'], 'test.png', { type: 'image/png' });
    const fileInput = getByTestId('image-uploader').querySelector('input[type="file"]');
    fireEvent.change(fileInput, { target: { files: [file] } });
    fireEvent.submit(getByTestId('form-image-uplaod'));
  })
})
