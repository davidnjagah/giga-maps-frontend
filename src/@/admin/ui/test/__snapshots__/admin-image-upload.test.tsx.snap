// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Us image upload Render MainAboutUsView 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: end;
}

.c0 .cds--search-magnifier svg {
  fill: #222;
}

<div
    class="c0"
  >
    <button
      class="cds--btn cds--btn--primary"
      type="button"
    >
      Upload New Image
      <svg
        aria-hidden="true"
        class="cds--btn__icon"
        fill="currentColor"
        focusable="false"
        height="16"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 9L3.7 9.7 7.5 5.9 7.5 15 8.5 15 8.5 5.9 12.3 9.7 13 9 8 4zM3 4V2h10v2h1V2c0-.6-.4-1-1-1H3C2.4 1 2 1.4 2 2v2H3z"
        />
      </svg>
    </button>
  </div>
  .c4 {
  padding: 1rem 1rem 1.5rem 1rem;
}

.c3 >tr >td {
  background: #FFFFFF;
  color: #474747;
  font-family: Open Sans;
  font-size: 0.875rem;
  font-style: normal;
  font-weight: 400;
  line-height: 1.25rem;
  letter-spacing: 0.01rem;
}

.c5 {
  cursor: pointer;
}

.c2 >tr >th {
  border-top: #E0E0E0;
  border-bottom: #E0E0E0;
  background: #F2F2F2;
}

.c0 {
  max-height: calc(100vh - 13rem);
}

.c1 {
  min-height: calc(100vh - 7rem);
}

.c1 .cds--data-table thead {
  position: sticky;
  top: 0;
  z-index: 1;
}

.c1 .cds--data-table-content {
  overflow-x: initial;
}

<div
    class="scrollbar-container c0 ps"
  >
    <div
      class="c1"
    >
      <div
        class="cds--data-table-content"
      >
        <table
          aria-label="user-roles-data-table"
          class="cds--data-table cds--data-table--lg"
        >
          <thead
            class="c2"
          >
            <tr>
              <th
                scope="col"
              >
                <div
                  class="cds--table-header-label"
                >
                  Name
                </div>
              </th>
              <th
                scope="col"
              >
                <div
                  class="cds--table-header-label"
                >
                  Image Link
                </div>
              </th>
              <th
                scope="col"
              >
                <div
                  class="cds--table-header-label"
                >
                  Action
                </div>
              </th>
            </tr>
          </thead>
          <tbody
            aria-live="polite"
            class="c3"
          >
            <tr>
              <td
                class="c4"
              >
                Banner Connectivity first section
              </td>
              <td
                class="c4"
              >
                <a
                  class="cds--link"
                  href="http://localhost:8000/media/images/f01448bb-9777-4ab8-9fdd-0a5b22887d45.png"
                  rel="noopener"
                  target="_blank"
                >
                  http://localhost:8000/media/images/f01448bb-9777-4ab8-9fdd-0a5b22887d45.png
                </a>
              </td>
              <td
                class="c4"
              >
                <span
                  data-testid="delete-about-us-image19"
                  style="margin-left: 10px;"
                >
                  <svg
                    aria-hidden="true"
                    class="c5"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 12H14V24H12zM18 12H20V24H18z"
                    />
                    <path
                      d="M4 6V8H6V28a2 2 0 002 2H24a2 2 0 002-2V8h2V6zM8 28V8H24V28zM12 2H20V4H12z"
                    />
                  </svg>
                </span>
              </td>
            </tr>
            <tr>
              <td
                class="c4"
              >
                First Resource
              </td>
              <td
                class="c4"
              >
                <a
                  class="cds--link"
                  href="http://localhost:8000/media/images/3cf1fa04-c3d5-469e-953d-af013ac99148.png"
                  rel="noopener"
                  target="_blank"
                >
                  http://localhost:8000/media/images/3cf1fa04-c3d5-469e-953d-af013ac99148.png
                </a>
              </td>
              <td
                class="c4"
              >
                <span
                  data-testid="delete-about-us-image20"
                  style="margin-left: 10px;"
                >
                  <svg
                    aria-hidden="true"
                    class="c5"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 12H14V24H12zM18 12H20V24H18z"
                    />
                    <path
                      d="M4 6V8H6V28a2 2 0 002 2H24a2 2 0 002-2V8h2V6zM8 28V8H24V28zM12 2H20V4H12z"
                    />
                  </svg>
                </span>
              </td>
            </tr>
            <tr>
              <td
                class="c4"
              >
                Second Resource
              </td>
              <td
                class="c4"
              >
                <a
                  class="cds--link"
                  href="http://localhost:8000/media/images/173e890f-b942-4e32-8c57-a2eac8a4fb93.png"
                  rel="noopener"
                  target="_blank"
                >
                  http://localhost:8000/media/images/173e890f-b942-4e32-8c57-a2eac8a4fb93.png
                </a>
              </td>
              <td
                class="c4"
              >
                <span
                  data-testid="delete-about-us-image21"
                  style="margin-left: 10px;"
                >
                  <svg
                    aria-hidden="true"
                    class="c5"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 12H14V24H12zM18 12H20V24H18z"
                    />
                    <path
                      d="M4 6V8H6V28a2 2 0 002 2H24a2 2 0 002-2V8h2V6zM8 28V8H24V28zM12 2H20V4H12z"
                    />
                  </svg>
                </span>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="ps__rail-x"
      style="left: 0px; top: 0px;"
    >
      <div
        class="ps__thumb-x"
        style="left: 0px; width: 0px;"
        tabindex="0"
      />
    </div>
    <div
      class="ps__rail-y"
      style="top: 0px; left: 0px;"
    >
      <div
        class="ps__thumb-y"
        style="top: 0px; height: 0px;"
        tabindex="0"
      />
    </div>
  </div>
  .c0 >.cds--modal-container {
  background: #383838;
  width: 38rem;
  height: auto;
  background: #fff;
}

.c0 >.cds--modal-container button svg {
  fill: #222!important;
}

.c0 >.cds--modal-container .cds--modal-close:hover {
  background-color: transparent!important;
}

.c1 >h3 {
  color: #161616;
  font-weight: 500;
}

.c1 button svg {
  fill: #fff;
}

.c1 .giga-text {
  display: flex;
  color: #fff;
  font-size: 2.4rem;
  font-weight: bolder;
}

.c1 .giga-text span:first-child {
  font-weight: 100;
}

.c1 .cds--modal-close:hover {
  background-color: #181818;
}

<div
    aria-hidden="true"
    class="cds--modal c0"
    id="import-csv-filter-modal"
    role="presentation"
  >
    <div
      aria-modal="true"
      class="cds--modal-container"
      role="dialog"
    >
      <button
        class="cds--visually-hidden"
        type="button"
      >
        Focus sentinel
      </button>
      <div
        class="cds--modal-container-body"
      >
        <div
          class="cds--modal-header c1"
        >
          <h3
            class="cds--modal-header__heading cds--type-beta"
          >
            Image Upload
          </h3>
          <button
            aria-label="Close"
            class="cds--modal-close"
            title="Close"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="cds--modal-close__icon"
              fill="currentColor"
              focusable="false"
              height="20"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17.4141 16L24 9.4141 22.5859 8 16 14.5859 9.4143 8 8 9.4141 14.5859 16 8 22.5859 9.4143 24 16 17.4141 22.5859 24 24 22.5859 17.4141 16z"
              />
            </svg>
          </button>
        </div>
      </div>
      <button
        class="cds--visually-hidden"
        type="button"
      >
        Focus sentinel
      </button>
    </div>
  </div>
</DocumentFragment>
`;
